coordinator/statement_executor.go

```diff
@@ -1039,10 +1039,31 @@ func (e *StatementExecutor) executeShowTagValues(ctx *query.ExecutionContext, q
 		return err
 	}
 
-	// Get all shards for all retention policies.
+	// If measurements include retention policies
+	// only look at those policies
+	rps := make([]string, 0, len(di.RetentionPolicies))
+	// Collect retention policies if specified
+	for _, m := range q.Sources.Measurements() {
+		if len(m.RetentionPolicy) > 0 {
+			rps = append(rps, m.RetentionPolicy)
+		}
+	}
+	// If no retention policies specified, use
+	// all retention policies
+	if len(rps) == 0 {
+		for _, rp := range di.RetentionPolicies {
+			rps = append(rps, rp.Name)
+		}
+	} else {
+		for _, rp := range rps {
```

Comment:
Should we deduplicate this list too?

Reply:
We do that in DedupInmemIndexes called from Store.TagValues()